docs/proposals/20201019-prometheus-rules-in-observatorium-api.md

```diff
@@ -0,0 +1,271 @@
+
+##### What if a user gives us an alerting rule
+
+Alerting rules have been explicitly de-scoped from this proposal as they require a story for managing alerting configuration and routing tenant alerts.
+
+However, Prometheus rules files permit recording and alerting rules to be freely mixed within the same file. The question arises - what do we do if a user provides us a file with mixed recording and alerting rules?
+
+###### 1. Reject Request
+
+One option would be to explicitly reject any requests that contains any `alert` rules in the payload. The user would be returned a `400 Bad Request` status code with a helpful warning like `Rules file rejected as it contains one or more alerting rules - alerting rules are not yet supported - please see the roadmap.`
+
+The downside here is that users may already have 'mixed' rules files, which they would need to split out in order to be accepted into the API.
+
+###### 2. Accept Request With Warning
+
+The other option would be to accept the rules request, store the recording rules and alerting rules alongside one another, but return a warning to saying `Warning: alerting rules are not currently support in Observatorium - please see the roadmap.`
+
+Under this option, we would also have to decide whether alerting rules are loaded into Rule and evaluated or only stored?
+
+###### Alert Decision
+
+My opinion is that the [least surprising](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) option for tenants is for their requests to be explicitly rejected with a clear error message.
+
+While this may impose some file splitting costs on the user's side, we avoid ambiguity such as valid questions such as:
+* Did all of my rules get stored or just some?
+* Are my alerting rules being evaluated or just the recording rules?
+
+Finally, warnings are easy to ignore especially if users are using automated process to interact with the API.
+
+#### Storage layer
+
+We will use a persistent storage mechanism that is external to the API deployment (i.e. no local storage). This enables us to satisfy our requirement of horizontally scaling the rule storage backend.
+
+While we have explicitly de-coupled the rule storage backend from the API, we require a concrete implementaion.
+
+We have a number of options for the storage layer backing the rules storage backend service.
+
+##### 1. ETCD
+
+One option would be to use the Kubernetes' control plane as the backing data store for tenant's recording rules.
+
+Pros:
+* Lowest possible overhead. No external storage is required to be orchestrated.
+* ETCD data model well suited for recording rules i.e. YAML.
+* Storage permissions are managed as Kubernetes objects.
+* We could leverage Recording Rules CRDs and get validation for free (?)
+
+Cons:
+* Implicitly ties Observatorium's implementation into a Kubernetes environment. Hard to back out.
```

Comment:
Really? What if we do this in this addon/plugin format so user can back it up with RBMS if needed?